File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    temp = [0]
    for n in numbers:
        arr = []
        for t in temp:
            arr.append(t+n)
            arr.append(t-n)
        temp = arr
        
    answer = 0
    for t in temp:
        if t == target:
            answer +=1
        
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let answer = 0;
    const len = numbers.length;
    
    function dfs(s, i) {
        if (i === len) {
            if (s === target) {
                answer++;
            }
            return;
        }
        
        dfs(s + numbers[i], i + 1); 
        dfs(s - numbers[i], i + 1);
    }
    
    dfs(0, 0);
    return answer;
}